contrib/filter-repo-demos/lint-history

```diff
@@ -59,6 +59,9 @@ example_text = '''CALLBACK
         def is_relevant(filename):
             BODY
 
+    Where filename is a fully qualified file name. The file is located in a temporary directory
+    that can be specified via TMPDIR environment variable.
```

Comment:
I think this will be somewhat confusing to users; they'll see it and expect "/Users/myusername/projects/clone-of-repo/path/to/code.c" instead of "path/to/code.c".  So we should probably instead mention that it is a path that is relative to the toplevel of the repository (as are all paths in filter-repo).

The first sentence and second sentences also seem jarring next to each other without more explanation.  The second sentence is unrelated to CALLBACKS and is true whether or not users specify the --relevant flag, so we should put that other sentence somewhere else; perhaps making a new section after "EXAMPLES" named "INTERNALS" or something.  And perhaps tweak the wording to make it a bit more clear, e.g. "Historic versions of files being linted will be written to a temporary directory before running the linting program; the location of this temporary directory can be controlled via the TMPDIR environment variable as per https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp."

Reply:
IIRC, in my test the file name was full (absolute), i.e. from the root directory, and the path didn't repeat the hierarchy from the project (i.e., `path/to/` in your example). Perhaps, it should be checked again.
Knowing path in the project is a good thing, since in some cases a transformations should be applied to some specific files in the project only.

Concerning TMPDIR — yes, agreed.